src/plugins/fastpath/src/java/org/jivesoftware/openfire/fastpath/commands/UpdateWorkgroup.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2007 Jive Software. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.openfire.fastpath.commands;
+
+
+import org.jivesoftware.xmpp.workgroup.Workgroup;
+import org.jivesoftware.xmpp.workgroup.WorkgroupManager;
+import org.jivesoftware.xmpp.workgroup.RequestQueue;
+import org.jivesoftware.xmpp.workgroup.Agent;
+import org.dom4j.Element;
+import org.jivesoftware.openfire.fastpath.util.WorkgroupUtils;
+import org.jivesoftware.openfire.commands.AdHocCommand;
+import org.jivesoftware.openfire.commands.SessionData;
+import org.jivesoftware.openfire.user.UserNotFoundException;
+import org.jivesoftware.util.StringUtils;
+import org.xmpp.forms.DataForm;
+import org.xmpp.forms.FormField;
+import org.xmpp.packet.JID;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command that allows to update existing workgroups.
+ *
+ * @author Gaston Dombiak
```

Comment:
This new code was written by Gato and copywrite to Jive 2007?

Reply:
Thanks. I will fix that